### ISSUE TITLE
What does uBlock do better than Safari Content Blockers?

### ISSUE THREAD

**Issue by vassudanagunta** [el1t/uBlock-Safari#112]
uBlock-Safari is falling behind uBlock Origin. It has 55 open issues on top of the 172 uBlock Origin issues, most of which probably apply to it as well.  Issues like #111 keep popping up.

Does uBlock Origin even make sense for Safari anymore since both the macOS and iOS versions support **Safari Content Blockers**, which I understand are implemented in the most efficient way possible?

It would help most if the experts (@gorhill and @el1t) weighed in.

**Comment by ghost** [comment]
Safari content blockers on both Mac/iOS are only limited to 50k rules maximum (for ads, trackers, etc).  Many content blockers do not inform the user of how often they update the block list either.  The major divide is the flexibility.  You can use uBlock in its default config and get along fine, or with a couple clicks you can block all 3rd party activity and take advantage of the "wide spectrum blocker" nature and take a whitelisting approach to your browsing.  You can also import different host files to your satisfaction and easily add your own custom filters and whitelists.  With a Content Blocker, if you visit a site that a content blocker doesn't handle well; you're probably out of luck.  They do not offer much dynamic filtering capabilities.  With uBlock, you can fine tune your experience on the fly without being subject to exterior resources.  

I've used uBlock on Safari for some time now and haven't come across a single show-stopping bug and it's only maintained by one guy.  I've tried content blockers out of curiosity and they have fantastic performance, but the implementation doesn't seem totally realized.  

My answer to your question is that unless the webkit team wants to greatly expand content blocking functionality, there is absolutely a place for this extension on Safari, and we're fortunate someone is maintaining it.  It's not my place to comment on issue #111 you highlighted because this software is provided as-is at @el1t's convenience until we hear otherwise.

**Comment by MysticalOS** [comment]
TLDR of above, not enough filter space, and no support for whitelisting at all.

For example I can tell you ublock rules actually break disquss on most sites without custom whitelisting since the way they use cross site cookies is not compatible with many filters. Heck, it's not even compatible with safari's built in cross site cookie block which requires disabling to use disquss and even some full fledged websites.

whitelisting is essencial to blocking and content blockers don't support it.

Down side of not using content blockers is some content doesn't get blocked fast enough so it still loads/consumes resources even if it's hidden by ublock. It's not a terrible idea to simply use a good content blocker on top of ublock to block things ublock can't. It'd be nice if someone could build a custom one that is specifically geared to fitting in  the most important network request filters into a 50k list, but I doubt that'll happen. I suspect most will wait or webkit to make feature less restrictive or just not support it.

**Comment by giusr** [comment]
The 50k rule limit seems arbitrary, if you look at 1Blocker X they have a simple workaround.

I still think uBlock Origin makes a lot of sense as an AddOn for Safari, what is really frustrating though, is that we just don't have a timeline for upgrades from the Safari Extension Gallery.

Some indication if this is just waiting for "app review" or if it depnds on something else would be nice.

**Comment by MysticalOS** [comment]
Just auto update from dev instead of gallery. it lets you choose update path

**Comment by Baccount** [comment]
Also 1Blocker X supports up to 350,000 block rules. This blog post explains it in more detail https://backstage.1blocker.com/say-hello-to-1blocker-x-8b55efe1ae36

**50,000 rules for each blocker and it uses 7 blockers**
![image](https://user-images.githubusercontent.com/10781882/39892238-d43178ee-5454-11e8-9e63-b8d3c4db63cf.png)

https://itunes.apple.com/us/app/1blocker-x/id1365531024?mt=8

@giusr 
@acasaceli

**Comment by gregforcey** [comment]
I'll admit the 1blocker approach to allowing more filters is a creative workaround. Too bad the 1blocker update is only for iOS and there is no intention to update the mac version.

**Comment by vassudanagunta** [comment]
looks like uBlock Origin Safari is dead.
